### PR TITLE
Add pre-commit.sh to the test.yml

### LINKED ISSUE #2176: Add precommit to GitHub Actions
- Add flag (--check?) to doc generating script to fail if file isn’t up to date
- Run pre-commit in GitHub Actions
- Everyone just remember to run pre-commit.sh

### PR DESCRIPTION
**Pull request by yiboliu**
This is to resolve #2176 by adding pre-commit.sh check to test.yml in workflows.

### Reasons for making this change
Add running pre-commit.sh before commit.  

### Related issues
#2176
#2566 

### Checklist

* [x] I've added a screenshot of the changes, if this is a frontend change
* [x] I've added and/or updated tests, if this is a backend change
* [x] I've run the [pre-commit.sh](https://github.com/codalab/codalab-worksheets/blob/master/pre-commit.sh) script
* [x] I've updated [docs](https://github.com/codalab/codalab-worksheets/tree/master/docs), if needed

### CODE REVIEW

File: .github/workflows/pre-commit.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+    # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+    - uses: actions/checkout@v2
+
+    # Runs a single command using the runners shell
+    - name: Run a one-line script
+      run: ./pre-commit.sh
```
Comment (percyliang) [inline .github/workflows/pre-commit.yml]: Pre-commit mutates the code, so need to check that it didn't change.  This also seems redundant with some of the checks already there...should we organize a bit better (cc @epicfaace)?

**Comment by epicfaace** [comment]
@yiboliu , issue #2176 is not about adding `pre-commit.yml`. We don't want to run precommit checks automatically upon every commit, actually, because the checks can take a long time to run.

That issue is just about adding a new check in the GitHub Actions `format` job, in which we want to run `./pre-commit.sh` and make sure that no files have been changed. Sorry if that wasn't clear!

**Comment by epicfaace** [comment]
@yiboliu can you please rename this PR and change the description (it might be better actually to just create a separate PR)? Additionally, you may need to run ./pre-commit.sh first to fix the formatting for CI.

File: pre-commit.sh [hunk 2]
```diff
@@ -5,8 +5,10 @@
 # Script fails immediately when any command fails.
 set -e
 
+python -m pip install --user virtualenv
+
```
Comment (epicfaace) [inline pre-commit.sh]: `python` might refer to Python 2 for many people's systems -- could you keep the original virtualenv setup on this file?

**Review by epicfaace — CHANGES_REQUESTED** [review]
Looks like the precommit change is failing. Can you run pre-commit.sh and then check in the changes so that the build passes?

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -30,8 +30,8 @@ jobs:
           key: pip-${{ hashFiles('requirements.dev.txt') }}
           restore-keys: |
             pip-
+      - run: ./pre-commit.sh && git diff --exit-code
       - run: pip install -r requirements.dev.txt
```
Comment (epicfaace) [inline .github/workflows/test.yml]: I think you can remove this line now.

**Review by epicfaace — CHANGES_REQUESTED** [review]

**Review by epicfaace — APPROVED** [review]

**Pull request merged**